client/history/client.go

```diff
@@ -1079,6 +1080,21 @@ func (c *clientImpl) GetFailoverInfo(
 	return c.client.GetFailoverInfo(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
 }
 
+func (c *clientImpl) RatelimitUpdate(ctx context.Context, request *types.RatelimitUpdateRequest, opts ...yarpc.CallOption) (*types.RatelimitUpdateResponse, error) {
+	if len(opts) == 0 {
```

Comment:
because there isn't a way to do so, yes.  can't == it, can't really reflect on it, nothing.

Reply:
WithShardKey is a CallOption: https://github.com/yarpc/yarpc-go/blob/v1.73.0/call.go#L70
Which is an encoding.CallOption: https://github.com/yarpc/yarpc-go/blob/v1.73.0/call.go#L34
Which is this struct: https://github.com/yarpc/yarpc-go/blob/dev/api/encoding/call_option.go#L29-L31
Which is implemented by this here: https://github.com/yarpc/yarpc-go/blob/dev/api/encoding/call_option.go#L60-L70

So our only real option (AFAICT) is to reflect on the internal `opt` field, and try to read the un-typed `field.String()` raw contents, and hope it's recognizable as a shard key (because reflection will not give us the actual type on private fields).

Reply:
poking a bit more, this works:
```go
reflect.ValueOf(k).FieldByName("opt").Elem().Type().Name() // shardKeyOption
```
and we should be able to detect it changing in tests, but tbh I'm not sure how reliable non-panicky that'd be with other args / future yarpc changes 🤔.

a "recover and log and fail" might take care of the edge cases we care about well enough, just feels like a potential bit of pain in a surprise-future.